platform/core/src/services/MeasurementService/MeasurementServiceAPI.js

```diff
@@ -0,0 +1,76 @@
+import log from '../../log';
+
+export default class MeasurementServiceAPI {
+  constructor({ service }) {
+    this.measurements = {};
+    this.listeners = {};
+
+    if (service) {
+      service.setServiceImplementation({
+        addOrUpdate: this.addOrUpdate.bind(this),
+        subscribe: this.subscribe.bind(this),
+      });
+    }
```

Comment:
We use `setServiceImplementation` in the UI Services because we can't know their implementation at time of instantiation. It also makes for a clearer "this is what you need to implement to fulfill this contract" as consumers seek to provide alternative implementations.

I'm unsure of if we need that separation for Measurements. I'll think on it a little more, and we can discuss this, but it may be simpler and may make more sense to provide the methods baked in. Swapping it out for a different implementation would mean matching the method signatures and using a different factory.

Reply:
Updated.